File: Python/ByTier/C/2131_C_Make_it_Equal.py
# Problem: CF 2131 C - Make it Equal
# https://codeforces.com/contest/2131/problem/C

"""
C. Make it Equal
time limit per test2 seconds
memory limit per test256 megabytes

Problem: 
Given two multisets S and T of size n and a positive integer k, determine if it is possible to transform S into T by repeatedly performing operations:
- Select an element x in S, remove one occurrence of x.
- Insert either x + k or |x - k| into S.

Approach:
1. For each element in both arrays, compute the minimum cost (number of operations) to make it congruent to 0 modulo k.
   This is done by taking min(x % k, k - x % k), representing the minimal steps needed to reach a multiple of k.
2. Sort these computed remainders for both arrays.
3. If sorted remainder arrays are equal, then S can be transformed into T.

Time Complexity: O(n log n) per test case due to sorting
Space Complexity: O(n) for storing the remainder arrays

Algorithms/Techniques:
- Modular arithmetic for equivalence classes
- Greedy approach using minimal operations
- Sorting for comparison of multiset properties
"""

def read_ints():
    """Read a line of integers from input."""
    return list(map(int, input().split()))


def can_make_equal(n, k, a, b):
    """
    Check if two arrays can be made equal by adding/subtracting multiples of k.
    Args:
        n: Number of elements in each array
        k: The modulo value for operations
        a: First array of integers
        b: Second array of integers
    Returns:
        bool: True if arrays can be made equal, False otherwise
    """
    # Compute minimal operations needed to make each element divisible by k
    remainders_a = [min(x % k, k - x % k) for x in a]
    remainders_b = [min(x % k, k - x % k) for x in b]
    # Compare sorted remainders
    return sorted(remainders_a) == sorted(remainders_b)


def main():
    test_cases = int(input())
    for _ in range(test_cases):
        n, k = map(int, input().split())
        a = list(map(int, input().split()))
        b = list(map(int, input().split()))
        if can_make_equal(n, k, a, b):
            print("YES")
        else:
            print("NO")


if __name__ == "__main__":
    main()


# https://github.com/VaHiX/codeForces/

File: Java/ByTier/C/2131_C_Make_it_Equal.java
// Problem: CF 2131 C - Make it Equal
// https://codeforces.com/contest/2131/problem/C

/*
C. Make it Equal

Purpose:
This program determines whether two multisets S and T can be made equal by applying a specific operation: 
for any element x in S, we can remove x and insert either (x + k) or |x - k| into S.
The key insight is that elements are effectively grouped by their remainder when divided by k,
and the transformation preserves the "distance" from multiples of k.

Algorithms/Techniques:
- Modular arithmetic to map numbers to their equivalence class based on remainder mod k
- Sorting and comparison of transformed arrays to check if multisets can be matched

Time Complexity: O(n log n) per test case due to sorting of arrays of size n.
Space Complexity: O(n) for storing the remainder arrays of size n.

*/

import java.util.*;
public class C_Make_it_Equal {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-- > 0) {
            int n = sc.nextInt();
            int k = sc.nextInt();
            int[] arr1 = new int[n];
            int[] arr2 = new int[n];
            for (int i = 0; i < n; i++) {
                int x = sc.nextInt() % k;
                // Calculate minimal distance to closest multiple of k
                arr1[i] = Math.min(x, (k - x) % k);
            }
            for (int i = 0; i < n; i++) {
                int x = sc.nextInt() % k;
                // Calculate minimal distance to closest multiple of k
                arr2[i] = Math.min(x, (k - x) % k);
            }
            Arrays.sort(arr1);
            Arrays.sort(arr2);
            // If the sorted arrays of distances are equal, multisets can be made equal
            if (Arrays.equals(arr1, arr2)) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
    }
}


// https://github.com/VaHiX/codeForces/